src/AppCommon/Infra/InteractiveHttpAuth.cs

```diff
@@ -23,11 +23,12 @@ public static async Task<HttpClient> CreateHttpClient(Uri authUri, int maxTries
 
             while (true)
             {
-                var httpHandler = new HttpClientHandler
+                var socketHandler = new SocketsHttpHandler
```

Comment:
I don't think we should ever be disposing of this regardless of exceptions. I think it should be passed into this method or have a global static one that is referenced. According to the [docs ](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-3.1#alternatives-to-ihttpclientfactory-2) it should live for the lifetime of the app. Maybe a static constructor in InteractiveHttpAuth could initialize a static property that is the SocketsHttpHandler? 

> There are alternative ways to solve the preceding problems using a long-lived [SocketsHttpHandler](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.socketshttphandler) instance.
 
> - Create an instance of SocketsHttpHandler when the app starts and use it for the life of the app.
> - Configure [PooledConnectionLifetime](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.socketshttphandler.pooledconnectionlifetime#system-net-http-socketshttphandler-pooledconnectionlifetime) to an appropriate value based on DNS refresh times.
> - Create HttpClient instances using new HttpClient(handler, disposeHandler: false) as needed.

Reply:
If we do the above, I'm not sure that we need the Func at all, but I could be missing something.

Reply:
It is effectively for the lifetime of the app, but it may take 3 tries to get it. In that case, only the 1st (default credentials, didn't work) and 2nd (you typed the wrong credentials the first time) get disposed. The 3rd sockets handler (where you typed the credentials right) gets captured by the closure of the Func<>, which goes into the Command's constructor, and is kept there for the rest of the app's lifetime.

But if (hypothetically) `InteractiveAuth.CreateHttpClientFactory(…)` were to be called again to (let's say) post things to Slack, that would need to create a new SocketsHttpHandler with credentials configured for that service.

Reply:
Ah ok, that explanation helps me understand a bit better why it is the way it is. I'm ok with it.